### PR TITLE
Game endscreen

### PR DESCRIPTION
**Pull request by Bobbelty**
# End of game screen

**1. End of game screen now pops up when the game is finished. It displays the following:**
- Win, loss or draw
- Your name vs Opponents name

This screen is a MUI dialog components and is placed over the match page. 

**2. Login refactoring**

Moved the login dropdown page to app.tsx. This means that the login window will slide down on every page of the application

**3. Other changes**
- Added two icons that replace 'W' and 'B' in the lobby that shows the colors of the players
- Moved themeprovider and cssbaseline (MUI components) to the root of the program which meant that it could be removed from every component that uses the same theme.
- Added error message to join page if the code does not match an existing game. This is displayed as a red message below the join button.
- Moved popup slide transition component to separate file in Util folders as multiple components now use the same transition.
- Small CSS fixes

### CODE REVIEW

File: src/Components/SetupGame/CreateLobby.tsx [hunk 1]
```diff
@@ -24,33 +18,32 @@ export default function CreateGame(props: { createGameFunction: any }) {
...
+                        type="submit"
+                        fullWidth
+                        variant="contained"
+                        sx={{ mt: 3, mb: 1, p: 1 }}
+                    >
+                        CREATE LOBBY
+                    </Button>
+                </Box>
+            </Paper>
+        </Container>
```
Comment (ValleSten) [inline src/Components/SetupGame/CreateLobby.tsx:46]: There is some weird bug here, the variant code is not validated the first time you create a lobby. That is, the first time you create a lobby you can leave the variant code blank without it complaining (you still can't start the game, as the code is invalid).  But if you play a game using a valid variant code until completion and then try to start another lobby with an invalid variant code, then the user will be notified. The bug is probably not here, I just marked the button as it is the entry point to the bug.

**Review by ValleSten — CHANGES_REQUESTED** [review]
Looks good! I found a weird bug though where the variation code is not validated the first time around, as described above.

The end screen looks very nice! Though it would be nice to have a button to request a rematch or to go back to the lobby instead of being forced back to the main menu. However, this could be regarded as a future feature and does not necessarily belong in this PR.

Please have a look at the validation bug and let me know if you are unable to recreate it. Then I'll be happy to accept this PR :)

File: src/Components/SetupGame/CreateLobby.tsx (hunk 1 above)
Comment (Bobbelty) [reply src/Components/SetupGame/CreateLobby.tsx:46]: I am not sure why the bug occurs and have no idea what I changed that could possibly affect that. I am however working on an improved create lobby page where the old system with a popup will be replaced. Instead, a red text will appear below the create button if the variant ID was invalid (like the join game ID code).

I think that a replay functionality of some kind is a great idea but should probably be regarded as a future feature.

Comment (ValleSten) [reply src/Components/SetupGame/CreateLobby.tsx:46]: Ah, I see, sounds good. I'll go ahead and accept this PR then :)

**Pull request merged**